### PR TITLE
Add option to set expiry date against clients

### PR DESCRIPTION
**Pull request by andrewhancox**

### CODE REVIEW

**Comment by djarran** [comment]
Hey Andrew, I've tested this on two local instances and this works just as expected. Thank you! We have just increased the version number to include the latest changes since the last version bump. 

Could you please microbump your version in `version.php` and `db/upgrade.xml` to reflect the latest version `2024030700` ?

**Comment by andrewhancox** [comment]
> Hey Andrew, I've tested this on two local instances and this works just as expected. Thank you! We have just increased the version number to include the latest changes since the last version bump.
> 
> Could you please microbump your version in `version.php` and `db/upgrade.xml` to reflect the latest version `2024030700` ?

That's now done.

**Comment by andrewhancox** [comment]
Has any thought been given to integrating this fix?

**Comment by andrewhancox** [comment]
Any thoughts about integrating this?

**Comment by bwalkerl** [comment]
Hi @andrewhancox ,

I haven't had a chance to look at this, but based on what @djarran has said we should be able to merge it if you resolve the conflicts again. Apologies for missing this earlier and making you do it twice.

**Comment by andrewhancox** [comment]
Done - no problem on the wait, will be good to be fully aligned with the main version.

File: version.php [hunk 1]
```diff
@@ -26,7 +26,7 @@
 defined('MOODLE_INTERNAL') || die();
 
 $plugin->component = 'mod_scormremote';
-$plugin->release = 2024092700;
+$plugin->release = 2024092701;
 $plugin->version = 2024092700; // Keep in lockstep with version.
```
Comment (bwalkerl) [inline version.php]: Just need to get CI green now. Looks like it's complaining about the version not matching release / upgrade steps.

Comment (andrewhancox) [reply version.php]: Was rushing, should be sorted now.

File: classes/form/client.php [hunk 2]
```diff
@@ -57,6 +57,8 @@ public function definition() {
         $mform->addRule('primarydomain', get_string('maximumchars', '', 255), 'maxlength', 255, 'server');
         $mform->addHelpButton('primarydomain', 'domain', 'mod_scormremote');
 
+        $mform->addElement('date_time_selector', 'expiry', get_string('expiry', 'mod_scormremote'), array('optional' => true));
```
Comment (bwalkerl) [inline classes/form/client.php:60]: One last thing - this string doesn't exist.

Comment (andrewhancox) [reply classes/form/client.php:60]: Missed that resolving the cherry pick... sorted now. Thanks for your patience!

**Pull request merged**